pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -2086,14 +2100,64 @@ impl Timeline {
         }
         // end: move the above part to the loop body
 
+        let mut rewrote_delta_layers = Vec::new();
+        if let Some(compaction_key_range) = &job_desc.partial_key_range {
+            for layer in job_desc.rewrite_layers {
```

Comment:
Would be nice to encode in the type system that `rewrite_layers` is only meaningful if `partial_key_range` is some.
Maybe by using a custom enum instead of Option for `partial_key_range`.

Eventually, partial compaction should be the only thing we do (not in this PR, branches are not yet supported)

Reply:
Just consolidated the code path for full+partial compaction so no more Option in the job desc 🤣